python/tvm/relax/op/transform.py

```diff
@@ -0,0 +1,282 @@
+
+    Returns
+    -------
+    result : relax.Expr
+        The reshaped result.
+
+    Note
+    ----
+    The ``-1`` inference is only performed at compile-time.
+    That is to say, in any case the dimension length of ``-1`` cannot be inferred in
+    compile-time, an error will be thrown.
+    """
+    return _ffi_api.reshape(data, new_shape)  # type: ignore
+
+
+def transpose(data: Expr, axes: Optional[List[int]] = None) -> Expr:
+    """Permutes the dimensions of an array.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input data to the operator.
+
+    axes : Optional[List[int]]
+        The target axes order, reverse order if not specified.
+
+    Returns
+    -------
+    result : relax.Expr
+        The transposed result.
+    """
+    return _ffi_api.transpose(data, axes)  # type: ignore
+
+
+def expand_dims(data: Expr, axis: Union[int, List[int]]) -> Expr:
+    """Insert new axes at the positions given by `axis`.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input data to the operator.
+
+    axis : Union[int, List[int]]
+        The axes at which the input array are expanded.
+        All values are required to lie in range `[-data.ndim - 1, data.ndim]`, with the convention
+        of negative indexing.
+
+    Returns
+    -------
+    result : relax.Expr
+        The transformed result.
+    """
+    if isinstance(axis, int):
+        axis = [axis]
+    return _ffi_api.expand_dims(data, axis)  # type: ignore
+
+
+def squeeze(data, axis: Optional[Union[int, List[int]]] = None) -> Expr:
+    """Squeeze axes in the array.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input data to the operator.
+
+    axis : Optional[Union[int, List[int]]
+        The set of axes to remove.
+        If axis = None, remove all axis of dimensions 1.
+        If any specified axis has dimension that does not equal 1, it is an error.
+
+    Returns
+    -------
+    result : relax.Expr
+        The squeezed result.
+    """
+    if isinstance(axis, int):
+        axis = [axis]
+    return _ffi_api.squeeze(data, axis)  # type: ignore
+
+
+def flatten(data: Expr) -> Expr:
+    """Flatten all the tensor dimensions into one.
+
+    Parameters
+    ----------
+    data : Expr
+        The input data to the operator.
+
+    Returns
+    -------
+    result : Expr
+        The flattened result.
+    """
+    return _ffi_api.flatten(data)  # type: ignore
+
+
+def cast(data: Expr, dtype: str) -> Expr:
+    """Cast input tensor to the given data type.
+
+    Parameters
+    ----------
+    data : relax.Expr
+        The input data to the operator.
+
+    dtype: str
+        The target data type
+
+    Returns
+    -------
+    result : relax.Expr
+        The casted result.
+    """
+    return _ffi_api.cast(data, dtype)  # type: ignore
+
+
+def full(
+    fill_value: Expr,
+    shape: Union[PrimExprLike, Tuple[PrimExprLike], Expr],
```

Comment:
According to the annotation, shape could be int or PrimExpr. Does that case need to be implemented in `MakeFull` and be test covered?

Reply:
Good catch. We shouldn’t allow single int or PrimExpr here. Will update the annotation with tests.